cluster-autoscaler/core/scale_up.go

```diff
@@ -371,24 +372,45 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 			Pods:      make([]*apiv1.Pod, 0),
 		}
 
+		if err := context.ClusterSnapshot.Fork(); err != nil {
+			klog.Errorf("Error while calling ClusterSnapshot.Fork; %v", err)
+			return &status.ScaleUpStatus{Result: status.ScaleUpError}, errors.ToAutoscalerError(errors.InternalError, err)
+		}
+
+		// add test node to snapshot
+		if err := context.ClusterSnapshot.AddNodeWithPods(nodeInfo.Node(), nodeInfo.Pods()); err != nil {
```

Comment:
What is this trying to achieve? You don't seem to be doing anything with the forked state, what's the point?

Reply:
Oops, missed it's only reverted in error branch, please ignore.